mbuild/tests/test_lattice.py

```diff
@@ -119,68 +118,260 @@ def test_lattice_spacings_negative_or_zero(self):
-        assert lat_2d.basis_vectors['test2'][0] == [.25, .25]
-        assert lat_3d.basis_vectors['test1'][0] == [0, 0, 0]
-        assert lat_3d.basis_vectors['test2'][0] == [.25, .25, .25]
+                                      lattice_vectors=None)
+        with pytest.raises(ValueError):
+            neg_lattice = mb.Lattice(neg_test3, dimension=3,
+                                     lattice_vectors=None)
+
+    def test_lattice_spacings_correct(self):
+        correct_spacings = [0.12, 0.13, 0.14]
+        input_spacings = [0.12, 0.13, 0.14]
+        test_lattice = mb.Lattice(input_spacings)
+        np.testing.assert_array_equal(test_lattice.lattice_spacings,
+                                      correct_spacings)
+
+    def test_basis_atoms_default(self):
+        space_1d = [1.]
+        space_2d = [1., 1.]
+        space_3d = [1., 1., 1.]
+        correct_name = 'default'
+
+        test_1d = mb.Lattice(space_1d, dimension=1, basis_atoms=None)
+        test_2d = mb.Lattice(space_2d, dimension=2, basis_atoms=None)
+        test_3d = mb.Lattice(space_3d, dimension=3, basis_atoms=None)
+
+        assert len(test_1d.basis_atoms.keys()) == 1
+        assert len(test_2d.basis_atoms.keys()) == 1
+        assert len(test_3d.basis_atoms.keys()) == 1
+        for lattice in test_1d, test_2d, test_3d:
+            for key in lattice.basis_atoms.keys():
+                assert key == correct_name
+
+            correct_pos = [0. for x in range(lattice.dimension)]
+            np.testing.assert_array_equal(lattice.basis_atoms[correct_name][0],
+                                          correct_pos)
+
+    def test_basis_atoms_input_type(self):
+        incorrect_types = [list(), tuple(), str()]
+        lat_space = [1., 1., 1.]
+        for the_type in incorrect_types:
+            with pytest.raises(TypeError):
+                test_lat = mb.Lattice(lat_space, basis_atoms=the_type)
+
+    def test_basis_atoms_input_size(self):
+        dim = 3
+        lat_space = [1., 1., 1.]
+        basis_too_large = {'test': [[0., 0., 0., 0.]]}
+        basis_too_large_mult = {'test': [[0., 0., 0., 0.], [0., 0., 0]]}
+        basis_too_small = {'test': [[0., 0.]]}
+        basis_too_small_mult = {'test': [[0., 0.], [0., 0., 0.]]}
+        basis_empty = {'test': [[]]}
+        basis_none = {'test': [[None, None, None]]}
+        to_test = [basis_too_large, basis_too_small, basis_empty, basis_none]
+
+        for errors in to_test:
+            with pytest.raises(ValueError):
+                test_lattice = mb.Lattice(lat_space, dimension=3,
+                                          basis_atoms=errors)
+
+    def test_basis_atoms_location(self):
+        dim = 3
+        lat_space = [1., 1., 1.]
+        basis_neg = {'test': [[0., 0., -.2]]}
+        basis_neg_mult = {'test': [[0., 0., 0.], [0., 0., -.2]]}
+        basis_positive = {'test': [[0., 0., 1.2]]}
+        basis_positive_mult = {'test': [[0., 0., 0.], [0., 0., 1.2]]}
```

Comment:
Also this general testing pattern where you're doing a bunch of different inputs lends itself quite nicely to pytest parameterize. Might be worth checking out and just grouping these all into a parameterized function so you get more fine grained error reports for the different scenarios.